Docker/directory-permission-setter/set_directory_permissions.sh

```
@@ -0,0 +1,26 @@
+# Split the space list into an array (space is chosen because of docker compose folded string literals)
+IFS=' ' read -r -a directories <<< "$DIRECTORIES_TO_SET"
+
+# Iterate over the array and apply chown
+for dir in "${directories[@]}"; do
+  # Ensure path starts with /
+  if [[ ! "$dir" =~ ^/ ]]; then
+    echo "Error: Path must be absolute: $dir"
+    exit 1
+  fi
+  # Prevent path traversal
+  if [[ "$path" =~ \.\. ]]; then
```

Shellcheck

[warning] path is referenced but not assigned.

(SC2154)